ext/mysql2/extconf.rb

```diff
@@ -69,7 +76,8 @@ def asplode lib
   libs = ['m', 'z', 'socket', 'nsl', 'mygcc']
   while not find_library('mysqlclient', 'mysql_query', lib, "#{lib}/mysql") do
     exit 1 if libs.empty?
-    have_library(libs.shift)
+    lib = libs.shift
+    asplode(lib) unless have_library(lib)
```

Comment:
I guess I meant - we *want* the loop to continue and only raise if none of the options were found.